src/main/scala/devices/debug/SBA.scala

```diff
@@ -273,11 +273,14 @@ class SBToTL(implicit p: Parameters) extends LazyModule {
     vecData := Vec.tabulate(16) { i => io.dataIn(8*i+7,8*i) }
     muxedData := vecData(counter)
 
+    // Need an additional check to determine if address is safe for Get/Put 
+    val rdLegal_addr = edge.manager.supportsGetSafe(io.addrIn, io.sizeIn)
+    val wrLegal_addr = edge.manager.supportsPutFullSafe(io.addrIn, io.sizeIn)
     val (rdLegal,  gbits) = edge.Get(0.U, io.addrIn, io.sizeIn)
     val (wrLegal, pfbits) = edge.Put(0.U, io.addrIn, io.sizeIn, muxedData)
 
-    io.rdLegal := rdLegal
-    io.wrLegal := wrLegal
+    io.rdLegal := rdLegal && rdLegal_addr
+    io.wrLegal := wrLegal && wrLegal_addr
```

Comment:
I'd ditch the 'rdLegal' and 'wrLegal' terms. They are redundant and a waste of hardware. Your new rdLegal_addr checks already cover both address and permissions checks.

FYI, if you know bounds on your io.sizeIn, you can get a better circuit by declaring that with the third option to supportsGetSafe/etc; ie:

edge.manager.supportsGetSafe(io.addrIn, io.sizeIn, TransferSizes(1 /* min possible in bytes */, 8? /* max possible */))